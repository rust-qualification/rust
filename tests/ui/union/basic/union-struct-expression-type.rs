//@ run-pass
// 2.3.1.
use std::any::type_name;
use std::mem;

fn type_of<T>(_: &T) -> &'static str {
    type_name::<T>()
}

#[repr(C)]
union MyUnion {
    x: i32,
    y: u32,
}

fn main() {
    let u = MyUnion { x: 2 };
    assert_eq!(mem::size_of::<MyUnion>(), mem::size_of::<u32>());
    assert_eq!(type_of(&u), "MyUnion");
}