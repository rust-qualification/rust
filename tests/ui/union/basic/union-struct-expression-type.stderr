warning: unused variable: `u`
  --> $DIR/union-struct-expression-type.rs:17:9
   |
LL |     let u = MyUnion { x: 2 };
   |         ^ help: if this is intentional, prefix it with an underscore: `_u`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: 1 warning emitted

